### PR TITLE
[MI-1754] dispatch fetch customer

### PR DESCRIPTION
**Pull request by joseconsador**
/cc @zendesk/mintegrations

### Description

Implement [redux-saga](https://github.com/redux-saga) to handle FETCH_CUSTOMER actions.
Split fetchApi to a separate module.
Added an ErrorNotification component, this component will be displayed when the `error` state is populated with values, see ErrorReducer for examples.

### For improvement
We can adapt a more robust pattern for sagas like in https://github.com/redux-saga/redux-saga/blob/master/examples/real-world/sagas/index.js 

### References
* JIRA: https://zendesk.atlassian.net/browse/MI-1754

### Risks
* [low] Uncaught API responses may produce unknown behavior.

### CODE REVIEW

File: src/Reducers/CustomerReducer.js [hunk 1]
```diff
@@ -2,14 +2,20 @@ import {
...
 } from '../Actions/constants.js';
 
-const customer = (state = {}, action) => {
+const customer = (customerState = {}, action) => {
   switch (action.type) {
-    case FETCH_CUSTOMER_SUCCESS:
-      return Object.assign({}, state, {
-        customer: action.customer,
+    case FETCH_CUSTOMER_SUCCESS: {
+      if (action.response.customers.length === 0) {
```
Comment (kcasas) [inline src/Reducers/CustomerReducer.js]: Why not `!action.response.customers`?

Comment (joseconsador) [reply src/Reducers/CustomerReducer.js]: @kcasas Yeah that's possible too, but `action.response.customers.length` looks more readable to me.

File: src/Reducers/ErrorReducer.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import {
+  FETCH_CUSTOMER_SUCCESS,
+} from '../Actions/constants.js';
+
+const errorReducer = (state = {}, action) => {
+  let newState = {};
+  switch (action.type) {
+    case FETCH_CUSTOMER_SUCCESS:
```
Comment (kcasas) [inline src/Reducers/ErrorReducer.js]: Seems a bit odd for a reducer with a purposes of catching errors to be listening on a success action. Isn't there any other way to refactor this?

**Comment by joseconsador** [comment]
@kcasas 💇‍♂️

File: src/Reducers/ErrorReducer.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import {
+  ADD_ERROR,
+  REMOVE_ERROR,
+} from '../Actions/constants.js';
+
+const errorReducer = (state = {}, action) => {
+  const newState = state;
```
[the remark this answers is not in the document]
Comment (joseconsador) [reply src/Reducers/ErrorReducer.js]: @samgavinio 💇‍♂️

**Pull request merged**